dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -60,49 +64,74 @@
 <script lang="ts">
 import { defineComponent } from "vue";
 import PrimeButton from "primevue/button";
-import { CompanyReport } from "@clients/backend";
 import ReportFormElement from "@/components/forms/parts/ReportFormElement.vue";
 import ElementsDialog from "@/components/general/ElementsDialog.vue";
 import { ReportToUpload, StoredReport } from "@/utils/FileUploadUtils";
 import UploadDocumentsForm from "@/components/forms/parts/elements/basic/UploadDocumentsForm.vue";
+import { YesNoFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import { CompanyReport } from "@clients/backend";
+import { ObjectType } from "@/utils/UpdateObjectUtils";
 
 export default defineComponent({
   name: "UploadReports",
+  inheritAttrs: false,
+  inject: {
+    injectReferencedReportsForPrefill: {
+      from: "referencedReportsForPrefill",
+      default: {},
+    },
+  },
   components: {
     UploadDocumentsForm,
     ReportFormElement,
     PrimeButton,
   },
-  emits: ["referenceableReportNamesChanged"],
+  emits: ["reportsUpdated"],
   data() {
     return {
       reportsToUpload: [] as ReportToUpload[] | undefined,
       storedReports: [] as StoredReport[],
     };
   },
   props: {
+    ...YesNoFormFieldProps,
     referencedReportsForPrefill: {
       type: Object as () => { [key: string]: CompanyReport },
     },
-  },
-  computed: {
-    allReferenceableReportNames(): string[] {
-      const namesOfFilesToUpload = this.reportsToUpload.map((reportToUpload) => reportToUpload.fileNameWithoutSuffix);
-      const namesOfStoredReports = this.storedReports.map((storedReport) => storedReport.reportName);
-      return namesOfFilesToUpload.concat(namesOfStoredReports);
+    isEuTaxonomy: {
+      type: Boolean,
+      default: false,
     },
   },
   watch: {
     referencedReportsForPrefill() {
       this.prefillAlreadyUploadedReports();
     },
   },
+  computed: {
+    namesOfFilesToUpload() {
```

Comment:
Should be "namesOfReportsToUpload" to avoid confusion

Reply:
Done.

Reply:
👍